src/bosh-director/lib/bosh/director/custom_middleware_collector.rb

```diff
@@ -0,0 +1,14 @@
+require 'prometheus/middleware/collector'
+
+module Bosh
+  module Director
+    class CustomMiddlewareCollector < Prometheus::Middleware::Collector
+      def strip_ids_from_path(path)
+        path
+          .gsub(%r{/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}(/|$)}, '/:uuid\\1')
+          .gsub(%r{/\d+(/|$)}, '/:id\\1')
+          .gsub(%r{/deployments/[a-z0-9_-]+(/|$)}, '/deployments/:deployment\\1')
```

Comment:
Is there a reason that the default behavior of `strip_ids_from_path` is copied from the superclass? Are you particularly concerned that the behavior of the superclass will change in the future? If not, this should be changed to call the superclass, then chain your `deployment` stripping code. (This also happens to be the way recommended by the docs: <https://github.com/prometheus/client_ruby/blob/master/examples/rack/README.md#collector>)

I bring this up because -prior to this change- we were using the default behavior of the `Prometheus::Middleware::Collector`'s `strip_ids_from_path` function... so if we ever upgraded that library and that function's behavior ever changed, we would automatically get the new code.